hazelcast/src/main/java/com/hazelcast/map/impl/tx/TxnDeleteBackupOperation.java

```diff
@@ -18,22 +18,44 @@
 
 import com.hazelcast.map.impl.MapDataSerializerHook;
 import com.hazelcast.map.impl.operation.RemoveBackupOperation;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
 import com.hazelcast.nio.serialization.Data;
 
+import java.io.IOException;
+
 public class TxnDeleteBackupOperation extends RemoveBackupOperation {
 
+    private String transactionId;
+
     public TxnDeleteBackupOperation() {
     }
 
-    public TxnDeleteBackupOperation(String name, Data dataKey) {
+    public TxnDeleteBackupOperation(String name, Data dataKey, String transactionId) {
         super(name, dataKey, false);
+        this.transactionId = transactionId;
     }
 
     @Override
     protected void runInternal() {
-        super.runInternal();
-
+        boolean exist = recordStore.existInMemory(dataKey);
+        recordStore.removeBackupTx(dataKey, getCallerProvenance(), transactionId);
         recordStore.forceUnlock(dataKey);
+        if (!exist) {
+            wbqCapacityCounter().decrement(transactionId, getPartitionId());
```

Comment:
Why do we decrement if it doesn't exist?

Reply:
It is to align with primary's delete operation, if no key we decrease counter.